internal/ingress/controller/template/template.go

```diff
@@ -1336,3 +1337,43 @@ func shouldLoadOpentracingModule(c interface{}, s interface{}) bool {
 
 	return false
 }
+
+func buildModSecurityForLocation(cfg config.Configuration, location *ingress.Location) string {
+	isMSEnabledInLoc := location.ModSecurity.Enable
+	isMSEnabled := cfg.EnableModsecurity
+
+	if !isMSEnabled && !isMSEnabledInLoc {
+		return ""
+	}
+
+	if !isMSEnabledInLoc {
+		return ""
+	}
+
+	var buffer bytes.Buffer
+
+	if !isMSEnabled {
+		buffer.WriteString(`modsecurity on;
+modsecurity_rules_file /etc/nginx/modsecurity/modsecurity.conf;
```

Comment:
this do have same problem that did exist before? check https://github.com/kubernetes/ingress-nginx/pull/4982/files

So what is the problem? in configmap modsecurity disabled, annotation modsecurity enabled and snippet added.

it is printing something like

```
modsecurity on;
modsecurity_rules_file /etc/nginx/modsecurity/modsecurity.conf;
modsecurity_rules '
%v
'
```

to config. However, this is the problem: values in modsecurity.conf cannot be overrided anymore with modsecurity_rules. modsecurity.conf contains SecRuleEngine DetectOnly, and I would like to override that with value `On`.

Reply:
> However, this is the problem: values in modsecurity.conf cannot be overrided anymore with modsecurity_rules.

So for me this combination is working with latest release i.e. the calls are getting blocked correctly,

```
    enable-modsecurity: "true"
    modsecurity-snippet: |
      Include /etc/nginx/owasp-modsecurity-crs/nginx-modsecurity.conf
      # SecRuleEngine On
      SecDebugLog /tmp/modsec_debug.log
```
(The second line must be getting parsed as comment I believe. Still keeping it as I haven't tested this by removing it.)

And in applications Ingress I have this annotations
```yaml
    nginx.ingress.kubernetes.io/modsecurity-snippet: |
      SecRuleEngine On
```

Somehow this works.